application/src/main/java/org/opentripplanner/routing/algorithm/filterchain/paging/PageCursorInputAggregator.java

```diff
@@ -0,0 +1,95 @@
+package org.opentripplanner.routing.algorithm.filterchain.paging;
+
+import java.util.function.Consumer;
+import org.opentripplanner.model.plan.paging.cursor.PageCursorInput;
+import org.opentripplanner.routing.algorithm.filterchain.filters.system.NumItinerariesFilter;
+import org.opentripplanner.routing.algorithm.filterchain.filters.transit.RemoveTransitIfStreetOnlyIsBetter;
+
+/**
+ * This class aggregates results from NumItinerariesFilter and RemoveTransitIfStreetOnlyIsBetter for PageCursorInput.
+ */
+public class PageCursorInputAggregator {
+
+  private final NumItinerariesFilter numItinerariesFilter;
+  private final RemoveTransitIfStreetOnlyIsBetter removeTransitIfStreetOnlyIsBetter;
+  private final Consumer<PageCursorInput> pageCursorInputSubscriber;
+
+  public static PageCursorInputAggregator.Builder of() {
+    return new Builder(new PageCursorInputAggregator());
+  }
+
+  private PageCursorInputAggregator() {
+    this.numItinerariesFilter = null;
+    this.removeTransitIfStreetOnlyIsBetter = null;
+    this.pageCursorInputSubscriber = null;
+  }
+
+  private PageCursorInputAggregator(Builder builder) {
+    this.numItinerariesFilter = builder.numItinerariesFilter();
+    this.removeTransitIfStreetOnlyIsBetter = builder.removeTransitIfStreetOnlyIsBetter();
+    this.pageCursorInputSubscriber = builder.pageCursorInputSubscriber();
+  }
+
+  public void createPageCursorInput() {
```

Comment:
I'm not sure if create is the right word here since this is not a factory method. Maybe `providePageCursorInput`?

Reply:
I changed this to `providePageCursorInput`